src/flat_file.cpp

```diff
@@ -392,11 +392,11 @@ static Status::type read_flat_file_tracy(const std::string& filename, Accelerato
         e.pass_method = PassMethod::pm_kickmap_pass;
         double tmpdbl; std::string filename;
         fp >> tmpdbl >> tmpdbl >> filename;
-        Status::type status = add_kicktable(filename, kicktable_list, e.kicktable_idx);
-        if (status == Status::success) {
+        unsigned int idx = add_kicktable(filename);
+        if (idx < 0) {
```

Comment:
is this condition correct?

Reply:
the condition is correct, @anacso17 . but your comment reminded that using 'unsigned int' as return type of add_kicktable is no appropriate: it turns out that c++ compiler treats unsigned int and int the same, for comparison reasons with other variable and with literals. only the bit representation changes.  for example, the following code works!

but  ```int``` instead of ```unsigned int``` is the correct type. i'll change. thanks!

```
#include <iostream>

int main() {
    int a = -1;
    unsigned int b = -1;
    if (b == -1) {
        std::cout << b << std::endl;
    }
    if (b == a) {
        std::cout << b << std::endl;
    }
}```

Reply:
you are right. the conditional at the Tracy/AT flat_file reader for kicktables is inverted. fixed!